### PR TITLE
fix: replace image tag

### LINKED ISSUE #2068: Cannot use colon with proxy port in predictor image name in inferenceservice-config
We changed the configuration of the predictor images in the ```inferenceservice-config``` ConfigMap (https://github.com/kserve/kserve/blob/master/config/configmap/inferenceservice.yaml) to use a proxy and a proxy port.

So e.g. ```"image": "ourproxy:8888/tensorflow/serving"```
instead of  ```"image": "tensorflow/serving"```

However using a colon in the predictor image name doesn't work because of ```strings.Split(image, ":")[0]```
in https://github.com/kserve/kserve/blob/922a9b7e8a9a86b5ae65faf4ce863927873fd456/pkg/controller/v1beta1/inferenceservice/utils/utils.go#L189

Can this please be changed to split the image name and the runtimeversion using the last colon in this string?
Or is there another way to configure a proxy for pulling these predictor images?

### PR DESCRIPTION
**Pull request by ittus**
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2068 

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?

### CODE REVIEW

File: pkg/controller/v1beta1/inferenceservice/utils/utils.go [hunk 1]
```diff
@@ -186,7 +187,12 @@ func ReplacePlaceholders(container *v1.Container, meta metav1.ObjectMeta) error
 func UpdateImageTag(container *v1.Container, runtimeVersion *string, isvcConfig *v1beta1.InferenceServicesConfig) {
 	image := container.Image
 	if runtimeVersion != nil && len(strings.Split(image, ":")) > 0 {
```
Comment (yuzisun) [inline pkg/controller/v1beta1/inferenceservice/utils/utils.go]: I think now the `len(strings.Split(image, ":")) > 0` check is redundant and can be removed.

**Comment by yuzisun** [comment]
@ittus Thanks for the fix! a small comment otherwise it looks good.

File: pkg/controller/v1beta1/inferenceservice/utils/utils.go (hunk 1 above)
Comment (ittus) [reply pkg/controller/v1beta1/inferenceservice/utils/utils.go]: Thanks. I removed it in https://github.com/kserve/kserve/pull/2074/commits/cc1397c3dc15990ef1c757513c42140ba32ac01e

**Comment by yuzisun** [comment]
Thanks @ittus for your contribution!!

/lgtm
/approve

**Pull request merged**